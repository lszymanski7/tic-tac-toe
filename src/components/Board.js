import React from 'react'
import PropTypes from 'prop-types'
import Square from './Square'

const Board = ({ animation, board, handleOnClick, winner }) => {
    /* istanbul ignore next */

    // Function âžž renders square on the board
    const renderSquare = (className, i) => {
        return (
            <Square
                animation={animation}
                className={className}
                handleOnClick={() => handleOnClick(i)}
                index={i}
                symbol={board[i]}
                winningLine={winner !== null ? winner.line.includes(i) : false}
            />
        )
    }

    return (
        <div className="board">
            <div className="board__row">
                {renderSquare(null, 0)}
                {renderSquare('right left', 1)}
                {renderSquare(null, 2)}
            </div>
            <div className="board__row">
                {renderSquare('top bottom', 3)}
                {renderSquare('top right bottom left', 4)}
                {renderSquare('top bottom', 5)}
            </div>
            <div className="board__row">
                {renderSquare(null, 6)}
                {renderSquare('right left', 7)}
                {renderSquare(null, 8)}
            </div>
        </div>
    )
}

Board.propTypes = {
    animation: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
    board: PropTypes.array.isRequired,
    handleOnClick: PropTypes.func.isRequired,
    winner: PropTypes.object
}

export { Board as default }
