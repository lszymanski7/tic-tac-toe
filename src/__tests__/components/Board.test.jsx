import React from 'react'
import { render, screen } from '@testing-library/react'
import Board from '../../components/Board'
import { ThemeProvider } from '../../contexts/ThemeContext'

const props = {
    animation: 'scale',
    board: Array(9).fill(null),
    handleOnClick: jest.fn(),
    winner: null
}

describe('Board', () => {
    it('should be rendered correctly.', () => {
        const { container } = render(<Board {...props} />, { wrapper: ThemeProvider })
        expect(container).toMatchSnapshot()
    })

    it('should have 9 squares.', () => {
        render(<Board {...props} />, { wrapper: ThemeProvider })
        const squares = screen.getAllByRole('button')
        expect(squares).toHaveLength(9)
    })
})
