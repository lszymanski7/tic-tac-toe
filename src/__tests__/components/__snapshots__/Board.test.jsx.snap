// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should be rendered correctly. 1`] = `
<div>
  <div
    class="board"
  >
    <div
      class="board__row"
    >
      <div
        class="square square--light"
      >
        <button
          aria-label="Square 0"
          class="square__button"
        />
      </div>
      <div
        class="square square--light right left"
      >
        <button
          aria-label="Square 1"
          class="square__button"
        />
      </div>
      <div
        class="square square--light"
      >
        <button
          aria-label="Square 2"
          class="square__button"
        />
      </div>
    </div>
    <div
      class="board__row"
    >
      <div
        class="square square--light top bottom"
      >
        <button
          aria-label="Square 3"
          class="square__button"
        />
      </div>
      <div
        class="square square--light top right bottom left"
      >
        <button
          aria-label="Square 4"
          class="square__button"
        />
      </div>
      <div
        class="square square--light top bottom"
      >
        <button
          aria-label="Square 5"
          class="square__button"
        />
      </div>
    </div>
    <div
      class="board__row"
    >
      <div
        class="square square--light"
      >
        <button
          aria-label="Square 6"
          class="square__button"
        />
      </div>
      <div
        class="square square--light right left"
      >
        <button
          aria-label="Square 7"
          class="square__button"
        />
      </div>
      <div
        class="square square--light"
      >
        <button
          aria-label="Square 8"
          class="square__button"
        />
      </div>
    </div>
  </div>
</div>
`;
